packages/@react-aria/checkbox/src/useCheckbox.ts

```diff
@@ -61,8 +62,25 @@ export function useCheckbox(props: AriaCheckboxProps, state: ToggleState, inputR
     }
   });
 
+  // Reset validation state on label click for checkbox with a hidden input.
+  let dispatch = () => {
+    // @ts-expect-error
+    let {[privateValidationStateProp]: groupValidationState} = props;
+
+    let {displayValidation, commitValidation} = groupValidationState
+    ? groupValidationState
+    : validationState;
+
+    if (displayValidation.isInvalid) {
+      commitValidation();
+    }
+  };
```

Comment:
Discussed with the team, we actually want it to match with the React Spectrum version of CheckboxGroup aka it gets re-invalidated if the user deselects the checkbox when the checkbox is required. This is a bit of an exception to the general "native" validation that is mentioned in the docs. 

That would mean you should be able just call `validationState.commitValidation()` here I believe. @devongovett I recall you mentioned something about the keyboard and mouse behavior not matching up as well but I forget the specifics, do you recall? If it was along the lines of having "Enter" properly commitValidation as well, then I think we'll need to pass that down to useToggle as well.

Additionally would you mind adding a test for this?

Reply:
@LFDanLu Will surely do, I was waiting on general guidance where this should head.

Would you mind expanding on the reasoning why this behavior is desired in RAC? As you already recognized it will create inconsistencies with the docs as well as other RACs or custom components utilizing form validation. I would be interested in finding out more here as its relevant to our component library and would I like to make a call whether to patch this if not relevant to us.

Also do you have guidance on what to do with #6096 ? Its very closely related and will remain open if this is an exception for CheckboxGroup only.